src/settings-ui/Settings.UI/OOBE/Views/VariantService.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using Microsoft.VariantAssignment.Client;
+using Microsoft.VariantAssignment.Contract;
+
+namespace Microsoft.PowerToys.Settings.UI.OOBE.Views
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Security.Cryptography;
+    using System.Text;
+    using System.Text.Json;
+    using Microsoft.PowerToys.Settings.UI.Library.Telemetry.Events;
+    using Microsoft.PowerToys.Telemetry;
+
+    public class VariantService
+    {
+        public void VariantAssignmentProvider_Initialize()
+        {
+            var vaSettings = new VariantAssignmentClientSettings
+            {
+                Endpoint = new Uri("https://default.exp-tas.com/exptas77/a7a397e7-6fbe-4f21-a4e9-3f542e4b000e-exppowertoys/api/v1/tas"),
+                EnableCaching = true,
+                ResponseCacheTime = TimeSpan.FromMinutes(5),
+            };
+
+            var vaClient = vaSettings.GetTreatmentAssignmentServiceClient();
+            var vaRequest = GetVariantAssignmentRequest();
+
+            try
+            {
+                var task = vaClient.GetVariantAssignmentsAsync(vaRequest);
+                var result = task.Result;
+                var allFeatureFlags = result.GetFeatureVariables();
+
+                var assignmentContext = result.GetAssignmentContext();
+                var featureNameSpace = allFeatureFlags[0].KeySegments[0];
+                var featureFlag = allFeatureFlags[0].KeySegments[1];
+                FeatureFlagValue = allFeatureFlags[0].GetStringValue();
+                PowerToysTelemetry.Log.WriteEvent(new OobeVariantAssignmentEvent() { Date = DateTime.Today, AssignmentContext = assignmentContext });
+            }
+            catch (Exception)
+            {
+                FeatureFlagValue = "current";
+            }
+        }
+
+        public string FeatureFlagValue { get; set; }
+
+        private static IVariantAssignmentRequest GetVariantAssignmentRequest()
+        {
+            string clientID = string.Empty;
+            string jsonFilePath = @"C:\Users\sophchen\Repos\PowerToys\src\settings-ui\Settings.UI.UnitTests\BackwardsCompatibility\TestFiles\v0.22.0\Microsoft\PowerToys\settings.json";
```

Comment:
I suspect this will only work on your machine ;)

Reply:
Oops! Good catch! I'll go ahead and fix that :)

Reply:
Just fixed! Shouldn't be hardcoded anymore.